app/components/chat_component.rb

```diff
@@ -0,0 +1,45 @@
+class ChatComponent < ViewComponent::Base
+  include Motion::Component
+
+  attr_reader :user, :conversation, :messages, :send_message_id
+
+  def initialize(user:, conversation:, messages:)
+    @user = user
+    @conversation = conversation
+    @messages = messages
+    @send_message_id = SecureRandom.uuid
+    self.class.stream_for(conversation, :handle_received)
+  end
+
+  map_motion :send_message
+  def send_message(event)
+    message = conversation.public_send(
+      ActiveAdmin::Chat.message_relation_name.pluralize
+    ).create!(
+      sender: user,
+      content: event.form_data[:message]
+    )
+
+    @send_message_id = SecureRandom.uuid
+
+    self.class.broadcast_to(conversation, ActiveAdmin::Chat::MessagePresenter.new(message))
```

Comment:
I would consider moving this broadcast to an `after_commit` on the message so that no matter how a message is created it automatically gets streamed into components.